src/utils/hook.ts

```diff
@@ -35,6 +35,12 @@ export function usePrevious<T>(value: T): T | undefined {
   return ref.current
 }
 
+export function useForkedState<S>(basedState: S): [S, Dispatch<SetStateAction<S>>] {
```

Comment:
It can be defined in storybook module because it's only used for stories

Reply:
It has a general capability, which is not related to the business of Storybook itself, and may be reused elsewhere in the future, so it is more appropriate to put it here.